crates/primitives/src/scripts/metadata.rs

```diff
@@ -0,0 +1,81 @@
+//! Primitives for Bridge metadata.
+
+use std::mem;
+
+use bitcoin::{key::constants::SCHNORR_PUBLIC_KEY_SIZE, XOnlyPublicKey};
+
+/// Execution Environment address size in bytes.
+const EE_ADDRESS_SIZE: usize = 20;
+
+/// u32 size in bytes.
+const U32_SIZE: usize = mem::size_of::<u32>();
+
+/// Metadata bytes that the Bridge uses to read information from the bitcoin blockchain and the
+/// sidesystem.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AuxiliaryData<'tag> {
+    /// Tag, also known as "magic bytes".
+    pub tag: &'tag [u8],
+
+    /// Deposit-specific metadata.
+    pub metadata: DepositMetadata,
+}
+
+/// Deposit-specific metadata that the Bridge uses to read information from the bitcoin blockchain
+/// and the sidesystem.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum DepositMetadata {
+    DepositRequestTx {
+        /// 32-bit X-only public key.
+        // TODO: make this a BOSD Descriptor.
+        takeback_pubkey: XOnlyPublicKey,
+
+        /// 20-byte Execution Environment address.
+        ee_address: Vec<u8>,
+    },
+    DepositTx {
+        /// Stake index.
+        ///
+        /// # Implementation Notes
+        ///
+        /// This is a 4-byte big-endian encoded unsigned 32-bit integer.
+        stake_index: u32,
+
+        /// 20-byte Execution Environment address.
+        ee_address: Vec<u8>,
+    },
+}
+
+impl<'tag> AuxiliaryData<'tag> {
+    /// Extracts the metadata as bytes.
+    pub fn to_vec(&'tag self) -> Vec<u8> {
+        let mut bytes = Vec::with_capacity(
+            self.tag.len() // FIXME: make this known at compile time once the tag length is defined.
+                + match &self.metadata {
+                    DepositMetadata::DepositRequestTx { .. } => SCHNORR_PUBLIC_KEY_SIZE + EE_ADDRESS_SIZE,
```

Comment:
You should get the size from the param.
```suggestion
                    DepositMetadata::DepositRequestTx { ee_address, .. } => SCHNORR_PUBLIC_KEY_SIZE + ee_address.len(),
```

Reply:
We can't it's now known at compile time. So we don't do `Vec::with_capacity()` and we do `Vec::new()`.

Reply:
Done in 7493aa70b37f98525e3e78e6697daca0ddf4659b